### PR TITLE
BaSP-MR-39: Class tests Put&Delete

### PR DESCRIPTION
**Pull request by Flor-Farruggia**
Add test for put, delete and get all empty from class entity.

### CODE REVIEW

File: src/tests/class.test.js [hunk 1]
```diff
@@ -1,5 +1,95 @@
...
+    expect(response.status).toBe(201);
+    expect(response.error).toBeFalsy();
+  });
+  test('For invalid ID', async () => {
+    const id = '64667748fc13ae7f027543d44';
+    const data = {
+      day: 'Sunday',
+    };
+    const response = await request(app).put(`/api/class/${id}`).send(data);
+    expect(response.status).toBe(500);
```
Comment (octavi01) [inline src/tests/class.test.js:25]: This should be a status 400

**Review by octavi01 — APPROVED** [review]

**Review by OscarLucero — APPROVED** [review]

File: src/tests/class.test.js [hunk 2]
```diff
@@ -1,5 +1,95 @@
...
-  expect(1 + 1).toBe(2);
+import request from 'supertest';
+import app from '../app';
+import Class from '../models/Class';
+import classSeed from '../seeds/class';
+
+beforeAll(async () => {
+  await Class.collection.insertMany(classSeed);
+});
+describe('PUT /api/class/:id', () => {
```
Comment (JuanMornik) [inline src/tests/class.test.js:9]: Missing status 500 test

File: src/tests/class.test.js [hunk 3]
```diff
@@ -1,5 +1,95 @@
...
+test('Server error. 500, internal server error', async () => {
+  jest.spyOn(Class, 'findByIdAndDelete').mockImplementation(() => {
+    throw new Error('Internal Server Error');
+  });
+  const response = await request(app).delete('/api/class/64667748fc13ae7f027543d4').send();
+  expect(response.status).toBe(500);
+  expect(response.error).toBeTruthy();
+});
+
+describe('GET /api/class', () => {
```
Comment (JuanMornik) [inline src/tests/class.test.js:83]: Missing status 500 test

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/tests/class.test.js (hunk 2 above)
Comment (JuanMornik) [reply src/tests/class.test.js:9]: Missing invalid ID test

File: src/tests/class.test.js [hunk 4]
```diff
@@ -1,5 +1,111 @@
...
+    jest.spyOn(Class, 'findByIdAndUpdate').mockImplementation(() => {
+      throw new Error('Internal Server Error');
+    });
+    const response = await request(app).put('/api/class/64667748fc13ae7f027543d4').send();
+    expect(response.status).toBe(500);
+    expect(response.error).toBeTruthy();
+  });
+});
+
+describe('DELETE /api/class/:id', () => {
```
Comment (JuanMornik) [inline src/tests/class.test.js:56]: Missing invalid ID test

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/tests/class.test.js [hunk 5]
```diff
@@ -57,6 +42,144 @@ describe('GET BY ID /api/class/:id', () => {
...
+    const response = await request(app).delete('/api/class/64667748fc13ae7f027543d9').send();
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
+  });
+  test('Invalid route. 404 wrong route.', async () => {
+    const response = await request(app).delete('/api/classes/64667748fc13ae7f027543d4').send();
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
+  });
+  test('Invalid id. 404 error, id not found.', async () => {
```
Comment (JuanMornik) [inline src/tests/class.test.js:132]: You have this test twice

File: src/tests/class.test.js [hunk 6]
```diff
@@ -57,6 +42,144 @@ describe('GET BY ID /api/class/:id', () => {
...
+    const response = await request(app).delete('/api/class/64667748fc13ae7f027543d9').send();
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
+  });
+  test('Get Invalid ID status 400', async () => {
+    const response = await request(app).delete('/api/class/64667748fc13ae7f027543d49999').send();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+  test('Invalid route. 404 wrong route.', async () => {
```
Comment (JuanMornik) [inline src/tests/class.test.js:142]: Delete this test too

File: src/tests/class.test.js [hunk 7]
```diff
@@ -57,6 +42,144 @@ describe('GET BY ID /api/class/:id', () => {
...
+    const response = await request(app).delete('/api/classes/64667748fc13ae7f027543d4').send();
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
+  });
+  test('Server error. 500, internal server error', async () => {
```
Comment (JuanMornik) [inline src/tests/class.test.js:147]: Delete this test

**Review by JuanMornik — CHANGES_REQUESTED** [review]

**Review by JuanMornik — APPROVED** [review]
